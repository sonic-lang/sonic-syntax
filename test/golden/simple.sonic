-- simply print 1

main :: IO () ;;
main = print 1
