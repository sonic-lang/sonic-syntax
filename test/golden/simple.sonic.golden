Right
    ( L
        { begin = Position
            { line = 3
            , column = 1
            }
        , content = Module
            ( Sequence
                [ L
                    { begin = Position
                        { line = 3
                        , column = 1
                        }
                    , content = TopDecl
                        ( L
                            { begin = Position
                                { line = 3
                                , column = 1
                                }
                            , content = WithAttrSet Nothing
                                ( L
                                    { begin = Position
                                        { line = 3
                                        , column = 1
                                        }
                                    , content = Simple
                                        ( Signature
                                            ( SignatureDecl
                                                { name = L
                                                    { begin = Position
                                                        { line = 3
                                                        , column = 1
                                                        }
                                                    , content = VarName "main"
                                                    , end = Position
                                                        { line = 3
                                                        , column = 6
                                                        }
                                                    }
                                                , type_ = L
                                                    { begin = Position
                                                        { line = 3
                                                        , column = 9
                                                        }
                                                    , content = Apply
                                                        ( L
                                                            { begin = Position
                                                                { line = 3
                                                                , column = 9
                                                                }
                                                            , content = Ctor
                                                                ( L
                                                                    { begin = Position
                                                                        { line = 3
                                                                        , column = 9
                                                                        }
                                                                    , content = Path
                                                                        { prefix = Nothing
                                                                        , modulePath = Nothing
                                                                        , name = L
                                                                            { begin = Position
                                                                                { line = 3
                                                                                , column = 9
                                                                                }
                                                                            , content = TyCtorName "IO"
                                                                            , end = Position
                                                                                { line = 3
                                                                                , column = 12
                                                                                }
                                                                            }
                                                                        }
                                                                    , end = Position
                                                                        { line = 3
                                                                        , column = 12
                                                                        }
                                                                    }
                                                                )
                                                            , end = Position
                                                                { line = 3
                                                                , column = 12
                                                                }
                                                            }
                                                        )
                                                        ( L
                                                            { begin = Position
                                                                { line = 3
                                                                , column = 12
                                                                }
                                                            , content = Tuple
                                                                ( L
                                                                    { begin = Position
                                                                        { line = 3
                                                                        , column = 12
                                                                        }
                                                                    , content = Sequence []
                                                                    , end = Position
                                                                        { line = 3
                                                                        , column = 15
                                                                        }
                                                                    }
                                                                )
                                                            , end = Position
                                                                { line = 3
                                                                , column = 15
                                                                }
                                                            }
                                                        )
                                                    , end = Position
                                                        { line = 3
                                                        , column = 15
                                                        }
                                                    }
                                                }
                                            )
                                        )
                                    , end = Position
                                        { line = 3
                                        , column = 15
                                        }
                                    }
                                )
                            , end = Position
                                { line = 3
                                , column = 15
                                }
                            }
                        )
                    , end = Position
                        { line = 3
                        , column = 15
                        }
                    }
                , L
                    { begin = Position
                        { line = 4
                        , column = 1
                        }
                    , content = TopDecl
                        ( L
                            { begin = Position
                                { line = 4
                                , column = 1
                                }
                            , content = WithAttrSet Nothing
                                ( L
                                    { begin = Position
                                        { line = 4
                                        , column = 1
                                        }
                                    , content = Simple
                                        ( Value
                                            ( ValueDecl
                                                { pat = L
                                                    { begin = Position
                                                        { line = 4
                                                        , column = 1
                                                        }
                                                    , content = Var
                                                        ( L
                                                            { begin = Position
                                                                { line = 4
                                                                , column = 1
                                                                }
                                                            , content = VarName "main"
                                                            , end = Position
                                                                { line = 4
                                                                , column = 6
                                                                }
                                                            }
                                                        )
                                                    , end = Position
                                                        { line = 4
                                                        , column = 6
                                                        }
                                                    }
                                                , body = L
                                                    { begin = Position
                                                        { line = 4
                                                        , column = 8
                                                        }
                                                    , content = Apply
                                                        ( L
                                                            { begin = Position
                                                                { line = 4
                                                                , column = 8
                                                                }
                                                            , content = Var
                                                                ( L
                                                                    { begin = Position
                                                                        { line = 4
                                                                        , column = 8
                                                                        }
                                                                    , content = Path
                                                                        { prefix = Nothing
                                                                        , modulePath = Nothing
                                                                        , name = L
                                                                            { begin = Position
                                                                                { line = 4
                                                                                , column = 8
                                                                                }
                                                                            , content = VarName "print"
                                                                            , end = Position
                                                                                { line = 4
                                                                                , column = 14
                                                                                }
                                                                            }
                                                                        }
                                                                    , end = Position
                                                                        { line = 4
                                                                        , column = 14
                                                                        }
                                                                    }
                                                                )
                                                            , end = Position
                                                                { line = 4
                                                                , column = 14
                                                                }
                                                            }
                                                        )
                                                        ( L
                                                            { begin = Position
                                                                { line = 4
                                                                , column = 14
                                                                }
                                                            , content = Literal
                                                                ( L
                                                                    { begin = Position
                                                                        { line = 4
                                                                        , column = 14
                                                                        }
                                                                    , content = Integer 1
                                                                    , end = Position
                                                                        { line = 5
                                                                        , column = 1
                                                                        }
                                                                    }
                                                                )
                                                            , end = Position
                                                                { line = 5
                                                                , column = 1
                                                                }
                                                            }
                                                        )
                                                    , end = Position
                                                        { line = 5
                                                        , column = 1
                                                        }
                                                    }
                                                , bindings = Nothing
                                                }
                                            )
                                        )
                                    , end = Position
                                        { line = 5
                                        , column = 1
                                        }
                                    }
                                )
                            , end = Position
                                { line = 5
                                , column = 1
                                }
                            }
                        )
                    , end = Position
                        { line = 5
                        , column = 1
                        }
                    }
                ]
            )
        , end = Position
            { line = 5
            , column = 1
            }
        }
    ,
        [ Comment
            { isMultiLine = False
            , content = " simply print 1"
            , location =
                ( Position
                    { line = 1
                    , column = 1
                    }
                , Position
                    { line = 1
                    , column = 18
                    }
                )
            }
        ]
    )